hunting/__main__.py

```diff
@@ -0,0 +1,276 @@
+@click.option('--format', 'output_format', default='toml', type=click.Choice(['toml', 'json'], case_sensitive=False),
+              help="Output format (toml or json).")
+@click.option('--query-only', is_flag=True, help="Only display the query content.")
+def view_hunt(uuid: str, path: str, output_format: str, query_only: bool):
+    """View a specific hunt by UUID or file path in the specified format (TOML or JSON)."""
+
+    # Get the hunt path or error message
+    hunt_path, error_message = get_hunt_path(uuid, path)
+
+    if error_message:
+        click.secho(f"{error_message}", fg="red", bold=True)
+        return
+
+    # Load the TOML data
+    hunt = load_toml(hunt_path)
+
+    # Handle query-only option
+    if query_only:
+        click.secho("Available queries:", fg="blue", bold=True)
+        # Format queries for display using tabulate and textwrap
+        table_data = [(i, textwrap.fill(query, width=120)) for i, query in enumerate(hunt.query)]
+        table_headers = ["Query"]
+        click.echo(tabulate(table_data, headers=table_headers, tablefmt="fancy_grid"))
+        return
+
+    # Output the hunt in the requested format
+    if output_format == 'toml':
+        click.echo(hunt_path.read_text())
+    elif output_format == 'json':
+        import json
+
+        # Convert the hunt object to a dictionary, assuming it's a dataclass
+        hunt_dict = hunt.__dict__
+        click.echo(json.dumps(hunt_dict, indent=4))
+
+
+@hunting.command('hunt-summary')
+@click.option('--breakdown', type=click.Choice(['platform', 'integration', 'language'], case_sensitive=False), default='platform',
+              help="Specify how to break down the summary: 'platform', 'integration', or 'language'.")
+def hunt_summary(breakdown: str):
+    """
+    Generate a summary of hunt queries, broken down by platform, integration, or language.
+    """
+    click.echo(f"Generating hunt summary broken down by {breakdown}...")
+
+    # Load all hunt queries
+    all_hunts = load_all_toml(HUNTING_DIR)
+
+    # Prepare data structures for summary
+    platform_summary = {}
+    integration_summary = {}
+    language_summary = {}
+
+    for hunt, path in all_hunts:
+        # Get the platform based on the folder name
+        platform = path.parent.parent.stem
+
+        # Extract integrations and languages used in the query
+        integrations = hunt.integration
+        languages = hunt.language
+
+        # Update platform-based summary
+        if platform not in platform_summary:
+            platform_summary[platform] = 0
+        platform_summary[platform] += 1
+
+        # Update integration-based summary
+        for integration in integrations:
+            if integration not in integration_summary:
+                integration_summary[integration] = 0
+            integration_summary[integration] += 1
+
+        # Update language-based summary
+        for language in languages:
+            if language == 'SQL':
```

Comment:
SQL is the query language used for OSQuery. To me it would be like saying "ES|QL, Discover" where discover is the tool but KQL is the actual language used to query the data behind the scenes.

Maybe we can add a note that if SQL, then it is meant for OSQuery. I think SQL is also universal to some degree and can be used for big data searches in Snowflake, BigQuery, etc. too depending on ingest of course.

Reply:
👍 This line just renders OSQuery in the stats summary table vs SQL.

Reply:
Updated the README so that we note within language explanation that if SQL is observed in TOML files or markdown, then this relates to OSQuery.